mars/services/scheduling/worker/workerslot.py

```diff
@@ -134,31 +143,48 @@ async def acquire_free_slot(self, session_stid: Tuple[str, str]):
         logger.debug('Slot %d acquired for subtask %r', slot_id, session_stid)
         raise mo.Return(slot_id)
 
-    def get_slot_address(self, slot_id: int):
-        return self._slot_control_refs[slot_id].address
-
-    def get_subtask_slot(self, session_stid: Tuple[str, str]):
-        return self._session_stid_to_slot.get(session_stid)
-
-    def release_free_slot(self, slot_id: int, pid: Optional[int] = None):
-        if pid is not None:
-            self._slot_to_proc[slot_id] = proc = psutil.Process(pid)
-            self._fresh_slots.add(slot_id)
-            # collect initial stats for the process
-            proc.cpu_percent(interval=None)
-
-        if slot_id in self._slot_kill_events:
-            event = self._slot_kill_events.pop(slot_id)
-            event.set()
+    def release_free_slot(self, slot_id: int, session_stid: Tuple[str, str]):
+        acquired_session_stid = self._slot_to_session_stid.pop(slot_id, None)
+        if acquired_session_stid is None:
+            raise SlotStateError(f'Slot {slot_id} is not acquired.')
+        if acquired_session_stid != session_stid:
+            raise SlotStateError(f'Slot {slot_id} releasing state incorrect, '
+                                 f'the acquired session_stid: {acquired_session_stid}, '
+                                 f'the releasing session_stid: {session_stid}')
+        acquired_slot_id = self._session_stid_to_slot.pop(acquired_session_stid)
+        assert acquired_slot_id == slot_id
 
-        session_stid = self._slot_to_session_stid.pop(slot_id, None)
-        self._session_stid_to_slot.pop(session_stid, None)
         logger.debug('Slot %d released', slot_id)
 
         if slot_id not in self._free_slots:
             self._free_slots.add(slot_id)
             self._semaphore.release()
 
+    def register_free_slot(self, slot_id: int, pid: int):
```

Comment:
Rename to `register_slot` as registered slots may not always be free.

Reply:
OK